app/java/src/main/java/com/google/cloudclientapi/ProcessingJob.java

```diff
@@ -14,26 +14,93 @@
  * limitations under the License.
  */
 
+package com.google.cloudclientapi;
 
-//
-// ProcessingJob.java - take data in raw_data bucket,
-//                      process, and store in processed bucket.
-//
+import com.google.gson.Gson;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-package com.google.cloudclientapi;
+public class ProcessingJob {
 
-abstract class ProcessingJob {
-  private static String RAW_DATA_BUCKET =
-      System.getenv().get("RAW_DATA_BUCKET");
-  private static String PROCESSED_DATA_BUCKET =
-      System.getenv().get("PROCESSED_DATA_BUCKET");
+  private static String RAW_DATA_BUCKET = System.getenv().get("RAW_DATA_BUCKET");
+  private static String PROCESSED_DATA_BUCKET = System.getenv().get("PROCESSED_DATA_BUCKET");
+  private static String RAW_DATA_FILE = "squirrels.csv";
+  private static final Logger logger = LoggerFactory.getLogger(CensusController.class);
 
   public static void main(String[] args) {
-    System.out.println(
+    logger.info(
+        String.format(
+            "🟢 Start ProcessingJob with: %s, %s", RAW_DATA_BUCKET, PROCESSED_DATA_BUCKET));
+    throwErrorIfInvalidArgs();
+    try {
+      String tempDataFilePath = downloadRawData();
+      Dictionary<String, SquirrelSegment> colorAgeLocationToCount =
+          processRawData(tempDataFilePath);
+      writeProcessedData(colorAgeLocationToCount);
+    } catch (Exception e) {
+      logger.error("ProcessingJob failed: " + e.getMessage());
+    }
+  }
+
+  private static void throwErrorIfInvalidArgs() {
+    if (RAW_DATA_BUCKET == null || RAW_DATA_BUCKET.isEmpty()) {
+      throw new IllegalArgumentException("RAW_DATA_BUCKET required");
```

Comment:
nit: should choose different exception to throw. `IllegalArgumentException` is intended for method arguments.